### PR TITLE
Remove old zk nodes structure

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

File: AppController/djinn.rb [hunk 1]
```diff
@@ -5675,21 +5672,27 @@ def fetch_revision(revision_key)
...
         Djinn.log_info("Waiting for a machine to have a copy of #{app_path}")
         Kernel.sleep(SMALL_WAIT)
         next
       end
 
-      ssh_key = remote_machine.ssh_key
-      ip = remote_machine.private_ip
+      # Get the ssh key to use for the remote machine.
+      ssh_key = ''
+      @nodes.each { |node| ssh_key = node.ssh_key if node.private_ip == ip }
```
Comment (cdonati) [inline AppController/djinn.rb]: I think `.find` is idiomatic here.

File: AppController/lib/zkinterface.rb [hunk 2]
```diff
@@ -130,12 +116,7 @@ def self.remove_revision_entry(revision_key, ip)
   end
 
   def self.get_revision_hosters(revision_key, keyname)
-    revision_hosters = get_children("#{ROOT_APP_PATH}/#{revision_key}")
-    converted = []
-    revision_hosters.each { |host|
-      converted << NodeInfo.new(get_job_data_for_ip(host), keyname)
-    }
-    converted
+    get_children("#{ROOT_APP_PATH}/#{revision_key}")
```
Comment (jeanleonov) [inline AppController/lib/zkinterface.rb]: `keyname` argument doesn't seem to be used any more.
Unless it's ruby best practice, could you also add explicit `return` statement?

Comment (obino) [reply AppController/lib/zkinterface.rb]: It's somewhat best practices ... but we also do both in that file ... I will add it since you asked nicely ROFL

**Comment by obino** [comment]
OCD build http://ocd.appscale.net:8080/job/Daily%20Build/6334/

File: AppController/djinn.rb [hunk 3]
```diff
@@ -5576,21 +5573,26 @@ def fetch_revision(revision_key)
...
         Djinn.log_info("Waiting for a machine to have a copy of #{app_path}")
         Kernel.sleep(SMALL_WAIT)
         next
       end
 
-      ssh_key = remote_machine.ssh_key
-      ip = remote_machine.private_ip
+      # Get the ssh key to use for the remote machine.
+      ssh_key = @nodes.find { |node| node["private_ip"] == ip }["ssh_key"]
+      if ssh_key.empty?
```
Comment (cdonati) [inline AppController/djinn.rb]: `ssh_key` will be `nil` if the IP address is not found. Also, trying to get `["ssh_key"]` from `nil` will raise an exception.

Comment (obino) [reply AppController/djinn.rb]: Good comments: thanks!

**Comment by obino** [comment]
OCD build http://ocd.appscale.net:8080/job/Daily%20Build/6622

**Pull request merged**